### PR TITLE
Make links clickable in rule classes documentation

### LINKED ISSUE #4656: The links in our rule descriptions should be clickables
## Expected Behavior
The rule description should have links in it's descriptions to give some "bibliography". Those links should be clickable in the documentation.

## Observed Behavior
[ObjectLiteralToLambda](https://detekt.dev/style.html#objectliteraltolambda) Right now they are not.

### PR DESCRIPTION
**Pull request by VitalyVPinchuk**
Fixes #4656 
Making the links clickable in the documentation of the rule classes as following:

Screenshot 2022-04-13 at 21 25 32

### CODE REVIEW

**Comment by cortinico** [comment]
Thanks for the PR @VitalyVPinchuk 
Have you built the website locally to verify that the links renders correctly? If yes, could post a screenshot?

**Comment by VitalyVPinchuk** [comment]
> Thanks for the PR @VitalyVPinchuk Have you built the website locally to verify that the links renders correctly? If yes, could post a screenshot?

Updated!

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseDataClass.kt [hunk 1]
```diff
@@ -28,7 +28,7 @@ import org.jetbrains.kotlin.types.KotlinType
  * Classes that simply hold data should be refactored into a `data class`. Data classes are specialized to hold data
  * and generate `hashCode`, `equals` and `toString` implementations as well.
  *
- * Read more about `data class`: https://kotlinlang.org/docs/data-classes.html
+ * Read more about [Data classes](https://kotlinlang.org/docs/data-classes.html)
```
Comment (schalkms) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseDataClass.kt:31]:
```suggestion
 * Read more about [data classes](https://kotlinlang.org/docs/data-classes.html)
```

**Review by schalkms — APPROVED** [review]
Looks good!

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseDataClass.kt (hunk 1 above)
Comment (VitalyVPinchuk) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseDataClass.kt]: I tried to make links description look exactly like their headings on web pages but ok, fixed.

**Review by cortinico — APPROVED** [review]

**Pull request merged**